### PR TITLE
Enable SYCL tests

### PR DESCRIPTION
**Pull request by rhornung67**
# Summary

- This PR enables SYCL tests in our current test set.

### CODE REVIEW

File: include/RAJA/policy/sycl/kernel/For.hpp
Comment (MrBurmark) [inline include/RAJA/policy/sycl/kernel/For.hpp:452]: diff_t is used in the hip and cuda backends where idx_type is used here.

File: include/RAJA/policy/sycl/kernel/ForICount.hpp
Comment (MrBurmark) [inline include/RAJA/policy/sycl/kernel/ForICount.hpp:69]: Should this be diff_t?
I worry about getting warnings about comparing signed and unsigned in `(i<len)`.

File: include/RAJA/policy/sycl/kernel/internal.hpp
Comment (MrBurmark) [inline include/RAJA/policy/sycl/kernel/internal.hpp:136]: Won't this cause remaining to be 0?

**Comment by artv3** [comment]
@homerdin , I just pushed some adjustments to the RAJA examples so we can build them when sycl is enabled. Since Sycl is a bit different than CUDA/HIP there are some code paths we need to guard for. Changes were mostly of the type `#if defined(RAJA_DEVICE_ACTIVE) -> defined(RAJA_ENABLE_HIP) || defined(RAJA_ENABLE_CUDA)`

**Comment by artv3** [comment]
I have some example improvements in https://github.com/LLNL/RAJA/pull/1273 for sycl.

**Comment by artv3** [comment]
We have some Docker sycl errors, I don't quite understand those though.

File: include/RAJA/policy/sycl/kernel/internal.hpp
Comment (homerdin) [reply include/RAJA/policy/sycl/kernel/internal.hpp:136]: When we are already at the max_work_group_size it will be 1.

**Comment by homerdin** [comment]
@rhornung67 I've reviewed the policies in RAJA User Guide and added SYCL atomics, everything is consistent now.

File: docs/sphinx/user_guide/feature/policies.rst
Comment (rhornung67) [inline docs/sphinx/user_guide/feature/policies.rst:664]: Is this last sentence true for SYCL also?

Comment (homerdin) [reply docs/sphinx/user_guide/feature/policies.rst:664]: I tested it and no it is not.  I will update the documentation.  When these tests are enabled, they compile with SYCL but fail at run time.

**Comment by rhornung67** [comment]
@homerdin I updated to a new dpcpp version in the CI, and resolved the merge conflicts in this. If all checks pass, and we get approvals, can it be merged? Or do you want to push more changes?

**Comment by rhornung67** [comment]
@homerdin still no luck with the multiple lambda tests when using intel 2022.1.0. I pushed a change to disable those for SYCL. If this passes now, we can circle back on SYCL tests and compiler versions after the release.

**Comment by rhornung67** [comment]
@MrBurmark @homerdin @artv3 @rchen20 @mdavis36 @trws This was now passing before I merged latest develop changes into it. This should be the last PR to merge before getting the RC branch merged into main for the release. Can we get some reviews?

I'm finishing the RC branch now.

File: include/RAJA/policy/sycl/reduce.hpp
Comment (mdavis36) [inline include/RAJA/policy/sycl/reduce.hpp:116]: We use an assigned sycl resource object here, and IIRC we don't need a "default" sycl resource like with CUDA / HIP because the Sycl backend handles the queue instantiation, right?

File: include/RAJA/policy/sycl/reduce.hpp
Comment (mdavis36) [inline include/RAJA/policy/sycl/reduce.hpp:613]: Why do minloc and maxloc produce race conditions here?

**Review by mdavis36 — APPROVED** [review]

File: include/RAJA/policy/sycl/reduce.hpp
Comment (homerdin) [reply include/RAJA/policy/sycl/reduce.hpp:116]: We use one because Sycl does not provide a default context so a newly constructed queue will be in a different context.  Intel provides an extension for a default context.

File: include/RAJA/policy/sycl/reduce.hpp
Comment (homerdin) [reply include/RAJA/policy/sycl/reduce.hpp:613]: Currently the location write can be disconnected with the min/max value write.

File: include/RAJA/policy/sycl/kernel/internal.hpp
Comment (MrBurmark) [reply include/RAJA/policy/sycl/kernel/internal.hpp:136]: In this else `remaining < launch_local.y` so won't `remaining / launch_local.y` always be 0?

Comment (homerdin) [reply include/RAJA/policy/sycl/kernel/internal.hpp:136]: `launch_local.y` is being assigned with the remaining value first.  line 136 could be changed to `remaining = 1;` as it ends up being `remaining / remaining`.  I wrote it this way to be more in line with the surrounding logic, but I can make the change if it makes things clearer.

**Comment by rhornung67** [comment]
@MrBurmark do you have any more concerns or questions about this? Or, are you OK to merge it?

File: include/RAJA/policy/sycl/kernel/internal.hpp
Comment (MrBurmark) [reply include/RAJA/policy/sycl/kernel/internal.hpp:136]: Oh, my bad, thanks for showing me that.

**Pull request merged**
